planner/core/expression_rewriter.go

```diff
@@ -1178,14 +1179,21 @@ func (er *expressionRewriter) Leave(originInNode ast.Node) (retNode ast.Node, ok
 }
 
 // newFunction chooses which expression.NewFunctionImpl() will be used.
-func (er *expressionRewriter) newFunction(funcName string, retType *types.FieldType, args ...expression.Expression) (expression.Expression, error) {
+func (er *expressionRewriter) newFunction(funcName string, retType *types.FieldType, args ...expression.Expression) (ret expression.Expression, err error) {
 	if er.disableFoldCounter > 0 {
-		return expression.NewFunctionBase(er.sctx, funcName, retType, args...)
+		ret, err = expression.NewFunctionBase(er.sctx, funcName, retType, args...)
 	}
 	if er.tryFoldCounter > 0 {
-		return expression.NewFunctionTryFold(er.sctx, funcName, retType, args...)
+		ret, err = expression.NewFunctionTryFold(er.sctx, funcName, retType, args...)
 	}
-	return expression.NewFunction(er.sctx, funcName, retType, args...)
+	ret, err = expression.NewFunction(er.sctx, funcName, retType, args...)
```

Comment:
why not `if ... else if ... else ...`? seems without `return`, the original path has been changed.

Reply:
Fixed.